client/src/actors/secure.rs

```diff
@@ -842,277 +827,260 @@ impl Handler<CallProcedure> for SecureClient {
     }
 }
 
-impl_handler!(
-procedures::SLIP10Generate, Result<crate::ProcResult, anyhow::Error>, (self, msg, _ctx), {
-
-    let key = if !self.keystore.vault_exists(msg.vault_id) {
+impl_handler!(procedures::SLIP10Generate, Result<crate::ProcResult, anyhow::Error>, (self, msg, _ctx), {
+    if !self.keystore.vault_exists(msg.vault_id) {
         let key = self.keystore.create_key(msg.vault_id);
-        self.db.init_vault(&key, msg.vault_id)?;
-        key
-    } else {
-        self.keystore.get_key(msg.vault_id).unwrap()
-    };
-
-    self.keystore.insert_key(msg.vault_id, key.clone());
+        self.db.init_vault(key, msg.vault_id)?;
+    }
+    let key = self.keystore.take_key(msg.vault_id).unwrap();
 
     let mut seed = vec![0u8; msg.size_bytes];
-    fill(&mut seed).map_err(|e| anyhow::anyhow!(e))?;
+    match fill(&mut seed) {
+        Ok(_) => {},
+        Err(e) => {
+            self.keystore.insert_key(msg.vault_id, key);
+            return Err(anyhow::anyhow!(e))
+        }
+    }
 
-    match self.db.write(&key, msg.vault_id, msg.record_id,&seed, msg.hint).map_err(|e| anyhow::anyhow!(e)) {
```

Comment:
good idea to get reduce the number of `anyhow` error conversions.

Reply:
agreed.